src/hooks/useSwapCallback.ts

```diff
@@ -662,42 +665,82 @@ export function useSwapCallback(
           { name: 'verifyingContract', type: 'address' },
         ]
 
-        const Swap = [
-          { name: 'amount0', type: 'uint256' },
-          { name: 'amount1', type: 'uint256' },
-          { name: 'path', type: 'address[]' },
-          { name: 'user', type: 'address' },
-          { name: 'nonce', type: 'uint256' },
-          { name: 'deadline', type: 'uint256' },
-          { name: 'feeAmount', type: 'uint256' },
+        const Forwarder = [
+          { name: 'from', type: 'address' },
           { name: 'feeToken', type: 'address' },
+          { name: 'maxTokenAmount', type: 'uint256' },
+          { name: 'deadline', type: 'uint256' },
+          { name: 'nonce', type: 'uint256' },
+          { name: 'data', type: 'bytes' },
+          { name: 'hashedPayload', type: 'bytes32' },
         ]
 
+        // const Swap = [
+        //   { name: 'amount0', type: 'uint256' },
+        //   { name: 'amount1', type: 'uint256' },
+        //   { name: 'path', type: 'address[]' },
+        //   { name: 'user', type: 'address' },
+        //   { name: 'deadline', type: 'uint256' },
+        // ]
+
         const domain = {
           name: 'ConveyorV2',
           version: '1',
           chainId: BigNumber.from(chainId).toHexString(),
           verifyingContract: CONVEYOR_V2_ROUTER_ADDRESS[chainId],
         }
 
-        const message = {
+        const payload = {
           amount0: BigNumber.from(amount0).toHexString(),
           amount1: BigNumber.from(amount1).toHexString(),
           path,
           user,
-          nonce: nonce.toHexString(),
           deadline: transactionDeadline.toHexString(),
-          feeAmount: BigNumber.from(feeOnTokenA.toFixed(0)).toHexString(),
+        }
+
+        const fnData = [
+          'function swapExactTokensForTokens(uint256 amount0,uint256 amount1,address[] path,address user,uint256 deadline)',
```

Comment:
The `swapExactTokensForTokens` function as defined in Solidity, see [code](https://github.com/automata-network/conveyor-v2/blob/d4146869bd77e1f3ba73c8ca4b1dd5c4a6a6f567/contracts/interfaces/IConveyorV2Router01.sol#L26-L28), such that it only takes in one parameter as `SWAP_TYPE`.

Solidity structs are defined as tuples within the ABI. That's why you can't simply pass these values as their individual arguments. In your syntax, the function would take 5 parameters instead of 1.